document_ai_warehouse/batch-ingestion-docai-wh/load_docs.py

```diff
@@ -0,0 +1,493 @@
+    document_schema_str = ""
+    if len(created_schemas) > 0:
+        document_schema_str = (
+            f"  - created document schema with id {','.join(list(created_schemas))}"
+        )
+    Logger.info(
+        f"Job Completed in {str(round(time_elapsed / 60))} minute(s): \n"
+        f"{document_schema_str}  \n"
+        f"  - processed gcs files={len(processed_files)} \n"
+        f"  - created dw documents={len(document_id_list)} \n"
+        f"  - processed gcs directories={len(processed_dirs)} \n"
+        f"  - created dw directories={len(created_folders)} \n"
+    )
+
+    if len(error_files) != 0:
+        Logger.info(
+            f"Following files could not be handled (Document page number exceeding limit of 200 pages? {','.join(error_files)}"
+        )
+
+
+def get_type(value: str):
+    if type(value) == bool or str(value) == "":
+        return "text_type_options"  # bool Not Supported
+    if is_date(value):
+        return "date_time_type_options"
+    if is_valid_int(value):
+        return "integer_type_options"
+    if is_valid_float(value):
+        return "float_type_options"
+
+    return "text_type_options"
+
+
+def is_valid_float(string: str):
+    try:
+        float(string)
+        return True
+    except ValueError:
+        return False
+
+
+def is_valid_bool(string: str):
+    return string.lower() in ["true", "false"]
+
+
+def is_valid_int(string: str):
+    return string.isdigit()
+
+
+def create_mapping_schema(display_name, names, options=True):
+    mapping_dic = {
+        "display_name": display_name,
+        "property_definitions": [],
+        "document_is_folder": False,
+        "description": "Auto-generated using batch upload",
+    }
+
+    if options:
+        for name, value in names:
+            definition = {
+                "name": name,
+                "display_name": name,
+                "is_repeatable": False,
+                "is_filterable": True,
+                "is_searchable": True,
+                "is_metadata": True,
+                "is_required": False,
+            }
+
+            v_type = get_type(value)
+            if v_type:
+                definition[v_type] = {}
+            mapping_dic["property_definitions"].append(definition)
+
+    file_path = os.path.join(
+        os.path.dirname(__file__), "schema_files", f"{display_name}.json"
+    )
+    with open(file_path, "w") as f:
+        json.dump(mapping_dic, f, indent=2)
+
+    return file_path
+
+
+def document_exists(reference_id: str) -> bool:
+    reference_path = f"referenceId/{reference_id}"
+    try:
+        dw_utils.get_document(reference_path, CALLER_USER)
+        return True
+    except NotFound:
+        return False
+
+
+def delete_document(reference_id: str):
+    Logger.info(f"delete_document reference_id={reference_id}")
+    reference_path = f"referenceId/{reference_id}"
+    dw_utils.delete_document(document_id=reference_path, caller_user_id=CALLER_USER)
```

Comment:
Why are these functions needed? Just seem to be wrappers around other functions

Reply:
Not sure, what would you suggest then?
Looks like a valid wrapper function for me